### PR TITLE
test(payment): INT-4926 increase code coverage for GooglePayStrategy

### PR DESCRIPTION
**Pull request by JoseAlvarezHdz**
## What?

Increase Code Coverage for googlepay-payment-strategy.ts

## Why?

To increase code coverage

### Before

Screen Shot 2021-11-11 at 14 21 24

### After

Screen Shot 2021-11-11 at 14 14 14

@bigcommerce/checkout @bigcommerce/payments @bigcommerce/apex-integrations

### CODE REVIEW

**Comment by JoseAlvarezHdz** [comment]
The only cases not covered were these two because there is no way `methodId` nor  `_googlePayOptions ` are undefined until that case, these are validated on previous methods

Screen Shot 2021-11-11 at 14 32 54

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts
Comment (vmparra) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:287]: Cool!

File: src/payment/strategies/googlepay/googlepay-adyenv2-payment-processor.ts
Comment (vmparra) [inline src/payment/strategies/googlepay/googlepay-adyenv2-payment-processor.ts:52]: Awesome!

https://media.giphy.com/media/xTiTnBMEz7zAKs57LG/giphy.gif

**Review by vmparra — APPROVED** [review]
Thanks for doing this @JoseAlvarezHdz , this is amazing.

Please make sure QA gives his blessing to this PR before merging.

cc @funivan

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts
Comment (leeBigCommerce) [reply src/payment/strategies/googlepay/googlepay-payment-strategy.ts:287]: Rather than set default values every time we de-structure, we could set them once within the `_displayWallet` function:

`private async _displayWallet(methodId: string, onPaymentSelect = noop, onError = noop): Promise<void> `

Or, we could avoid setting default values altogether and use optional execution syntax: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

```tsx
const runOptionalCallback = (optionalCallback?: () => void) => {
    return optionalCallback?.();
}

runOptionalCallback();

runOptionalCallback(() => console.log('I ran'));
```

**Review by leeBigCommerce — COMMENTED** [review]
Just incase this was missed: https://github.com/bigcommerce/checkout-sdk-js/pull/1288#discussion_r748917477

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts
Comment (JoseAlvarezHdz) [reply src/payment/strategies/googlepay/googlepay-payment-strategy.ts]: it makes sense to me, i will take you first option if its ok to you

**Review by leeBigCommerce — APPROVED** [review]
👍

**Comment by leeBigCommerce** [comment]
@bigcommerce/checkout code owner review, please 🔍

**Review by animesh1987 — APPROVED** [review]

**Comment by BC-AAcevedo** [comment]
:green_heart: tested
QA Details: GooglePay working as expected
[INT-4926 Increase code coverage on GooglePayPaymentStrategy.docx](https://github.com/bigcommerce/checkout-sdk-js/files/7645033/INT-4926.Increase.code.coverage.on.GooglePayPaymentStrategy.docx)

**Pull request merged**
